enhancements/storage/csi-inline-vol-security.md

```diff
@@ -0,0 +1,372 @@
+should deny admission.
+
+The following table demonstrates the pod admission behavior for a pod that
+mounts a CSI ephemeral volume with the given pod security enforcement profile
+and CSI ephemeral volume profile:
+
+| PodSecurity Profile | driver label: restricted | driver label: baseline | driver label: privileged |
+| ------------------- | ----------------- | --------------- | ----------------- |
+| restricted | Allowed | Denied  | Denied  |
+| baseline   | Allowed | Allowed | Denied  |
+| privileged | Allowed | Allowed | Allowed |
+
+#### Pod Security Profile Warning
+
+The `CSIVolumeAdmission` admission webhook can also provide user-facing warnings
+(via appropriate HTTP response headers) if the CSI driver’s effective profile
+is more permissive than the pod security warning profile for the pod namespace.
+
+The following table demonstrates the pod admission behavior with the given pod
+security warning profile and CSI driver effective profile:
+
+| PodSecurity Profile | driver label: restricted | driver label: baseline | driver label: privileged |
+| ------------------- | ----------------- | --------------- | ----------------- |
+| restricted | No Warning | Warning    | Warning  |
+| baseline   | No Warning | No Warning | Warning  |
+| privileged | No Warning | No Warning | No Warning |
+
+#### Pod Security Profile Audit
+
+The `CSIVolumeAdmission` admission webhook can also apply audit annotations to
+the pod if the CSI driver’s effective profile is more permissive than the pod
+security audit profile for the pod namespace.
+
+The following table demonstrates the pod admission behavior with the given pod
+security audit profile and CSI driver effective profile:
+
+| PodSecurity Profile | driver label: restricted | driver label: baseline | driver label: privileged |
+| ------------------- | ----------------- | --------------- | ----------------- |
+| restricted | No Audit | Audit    | Audit  |
+| baseline   | No Audit | No Audit | Audit  |
+| privileged | No Audit | No Audit | No Audit |
+
+#### Default Behavior
+
+If the referenced CSIDriver for a CSI ephemeral volume does not have the
+`csi-ephemeral-volume-profile` label, the `CSIVolumeAdmission` admission webhook
+will consider the driver to have the `privileged` profile for enforcement,
+warning, and audit behaviors. Likewise, if the pod’s namespace does not have
+the pod security admission label set, the admission webhook will assume the
+`restricted` profile is allowed for enforcement, warning, and audit decisions.
+
+This means that if no labels are set, CSI ephemeral volumes using that CSIDriver
+will _only_ be usable in privileged namespaces by default. This restrictive
+default policy prevents risky drivers from being used as inline volumes in
+unprivileged namespaces, unless the cluster administrator makes a conscious
+decision to allow it.
+
+Like other admission webhooks, this webhook will have its own configuration API
```

Comment:
I don't think you should offer this knob.  It is equivalent to a cluster-admin issuing `oc get csidriver -oname | oc label make-my-cluster-insecure`, right?

Reply:
That's fair, I'll remove this.